### PR TITLE
introduce OptionalDecoder

### PR DESCRIPTION
**Pull request by arielsegura**
Addresses #294

### CODE REVIEW

File: feign-optional-decoder/src/main/java/feign/optionals/OptionalDecoder.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import feign.Response;
+import feign.Util;
+import feign.codec.Decoder;
+
+import java.io.IOException;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.Optional;
+
+public class OptionalDecoder implements Decoder {
```
Comment (adriancole) [inline feign-optional-decoder/src/main/java/feign/optionals/OptionalDecoder.java]: final

File: feign-optional-decoder/src/main/java/feign/optionals/OptionalDecoder.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    final Decoder delegate;
+
+    public OptionalDecoder(Decoder delegate) {
+        this.delegate = delegate;
```
Comment (adriancole) [inline feign-optional-decoder/src/main/java/feign/optionals/OptionalDecoder.java]: check null

**Comment by arielsegura** [comment]
@adriancole done!

**Comment by spencergibb** [comment]
I like `feign-java8` as well

**Comment by arielsegura** [comment]
Naming it as feign-java8 will imply that further features around java8 will have to be shipped in this artefact, so for instance if one just wants to use Optional stuff, will bring the whole java8 support.

**Comment by arielsegura** [comment]
Fair enough 👍

**Comment by adriancole** [comment]
if you can rejig in the next couple days, will go in the next release. Don't forget CHANGES file
cc @jfuerth

**Comment by arielsegura** [comment]
Done!

**Comment by adriancole** [comment]
mind renaming the dir to java8 (pom artifact is good, just this is more conventional with others)

**Comment by arielsegura** [comment]
@adriancole renamed

**Pull request merged**